### PR TITLE
fix(application-system): Configure upload buckets, fix bull queue and improve FileUploadFormField

### PR DESCRIPTION
**Pull request by eirikurn**
## What

Specify what you're trying to achieve

## Why

Specify why you need to achieve this

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

**Review by oddsson — APPROVED** [review]

File: apps/application-system/api/src/app/modules/application/upload.processor.ts [hunk 1]
```diff
@@ -1,44 +1,58 @@
...
+    const destinationKey = `${applicationId}/${sourceKey}`
+    const url = await this.fileStorageService.copyObjectFromUploadBucket(
+      sourceKey,
       destinationBucket,
-      region,
-      bucket,
+      destinationKey,
     )
+
+    const resultUrl = `https://${destinationBucket}.s3-${this.config.region}.amazonaws.com/${destinationKey}`
```
Comment (dabbeg) [inline apps/application-system/api/src/app/modules/application/upload.processor.ts:46]: You already have the url from the `copyObjectFromUploadBucket`

File: apps/application-system/api/src/app/modules/application/application.configuration.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import { registerAs } from '@nestjs/config'
+
+export const applicationConfiguration = registerAs('application', () => ({
+  attachmentBucket: process.env.APPLICATION_ATTACHMENT_BUCKET,
```
Comment (dabbeg) [inline apps/application-system/api/src/app/modules/application/application.configuration.ts:4]: Any reason for not having this in the environments file?

File: apps/application-system/api/src/app/modules/application/upload.processor.ts (hunk 1 above)
Comment (eirikurn) [reply apps/application-system/api/src/app/modules/application/upload.processor.ts:46]: Good catch. I went back and forth on where the URLs should be managed.

File: apps/application-system/api/src/app/modules/application/application.configuration.ts (hunk 2 above)
Comment (eirikurn) [reply apps/application-system/api/src/app/modules/application/application.configuration.ts:4]: Good question. I guess I missed some discussions about how we configure NestJS modules and was just following the [configuration guide](https://docs.nestjs.com/techniques/configuration).

I'm not sure how I feel about the `Module.register(config)` pattern. It's non-standard, and requires top-level modules to provide configuration all sub-modules, which exposes implementation details, eg with nested modules.

On the other hand, I understand the benefit of having an overview of all configurations/environment-variables in one place.

Comment (eirikurn) [reply apps/application-system/api/src/app/modules/application/application.configuration.ts:4]: I'll change this to match the convention.

**Review by Rikkiabb — DISMISSED** [review]
Nice work 👏 Looks good from our side 💯

**Review by dabbeg — APPROVED** [review]
Nice :+1:

**Review by oddsson — APPROVED** [review]

**Review by Rikkiabb — APPROVED** [review]

**Review by jeremybarbet — APPROVED** [review]
LGTM

**Review by unnarb — APPROVED** [review]

**Pull request merged**